serde-support/src/main/java/io/micronaut/serde/support/deserializers/SpecificObjectDeserializer.java

```diff
@@ -63,7 +62,9 @@ public Object deserialize(Decoder decoder, DecoderContext decoderContext, Argume
                 final PropertiesBag<Object>.Consumer creatorParams = db.creatorParams.newConsumer();
                 final DeserBean.DerProperty<Object, Object> creator = creatorParams.getNotConsumed().iterator().next();
                 final Object val = deserializeValue(decoderContext, decoder, creator, creator.argument, null);
-                return db.introspection.instantiate(val);
+                Object[] args = new Object[1];
```

Comment:
initialise on the same line `new Object[] { val }`

Reply:
done via https://github.com/micronaut-projects/micronaut-serialization/pull/492